webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/ncf/TechniqueReader.scala

```diff
@@ -0,0 +1,85 @@
+package com.normation.rudder.ncf
+
+import better.files.File
+import com.normation.errors.IOResult
+import com.normation.errors.Inconsistency
+import com.normation.rudder.hooks.Cmd
+import com.normation.rudder.hooks.RunNuCommand
+import com.normation.rudder.rest.RestExtractorService
+import net.liftweb.common.EmptyBox
+import net.liftweb.common.Full
+import net.liftweb.json.JsonAST.JArray
+import net.liftweb.json.JsonAST.JObject
+import net.liftweb.json.parse
+import zio.ZIO
+
+class TechniqueReader(
+  restExtractor  : RestExtractorService
+) {
+
+
+  import zio.syntax._
+  import better.files.File._
+  val configuration_repository = root / "var" / "rudder" / "configuration-repository"
+  val methodsFile = configuration_repository / "ncf" / "generic_methods.json"
+  def getAllTechniqueFiles(currentPath : File): List[File]= {
+    val (subdirs, files) = currentPath.children.partition(_.isDirectory)
```

Comment:
this should be in an `IO`, because it's io plus is can block

Reply:
Everything should be a big IO